### PR TITLE
[UPDATED] Use Protobuf instead of JSON

### LINKED ISSUE #9: Don't use JSON marshaller
Use something more efficient.

### PR DESCRIPTION
**Pull request by kozlovic**
Use Protobuf for VoteRequest, VoteResponse and HearBeat.
In this PR the state is not yet using Protobuf.

Resolves #9

### CODE REVIEW

File: nats_rpc.go [hunk 1]
```diff
@@ -24,11 +25,12 @@ var (
 	NotInitializedErr = errors.New("graft(nats_rpc): Driver is not properly initialized")
 )
 
+// NatsRpcDriver is an implementation of the RPCDriver using NATS.
 type NatsRpcDriver struct {
 	sync.Mutex
 
-	// NATS encoded connection.
-	ec *nats.EncodedConn
```
Comment (derekcollison) [inline nats_rpc.go]: Can still use EC, they can do PBs..

File: nats_rpc.go [hunk 2]
```diff
@@ -107,30 +108,45 @@ func (rpc *NatsRpcDriver) vreqSubject() string {
 	return fmt.Sprintf(VOTE_REQ_SUB, rpc.node.ClusterInfo().Name)
 }
 
-// Heartbeat callback which will place the heartbeat on the Graft
+// HeartbeatCallback will place the heartbeat on the Graft
 // node's appropriate channel.
-func (rpc *NatsRpcDriver) HeartbeatCallback(hb *Heartbeat) {
+func (rpc *NatsRpcDriver) HeartbeatCallback(msg *nats.Msg) {
+	hb := &pb.Heartbeat{}
+	if err := hb.Unmarshal(msg.Data); err != nil {
```
Comment (derekcollison) [inline nats_rpc.go]: Would use an EC..

**Comment by kozlovic** [comment]
The PR failed because of the StateChange issue fixed in other PR, going to recycle:
```
./scripts/cov.sh TRAVIS
=== RUN   TestStateChangeHandler
--- FAIL: TestStateChangeHandler (0.62s)
	handler_test.go:51: Did not receive correct states for state change: &{From:Candidate To:Leader}
=== RUN   TestErrorHandler
--- PASS: TestErrorHandler (0.97s)
=== RUN   TestHeartBeatAsFollower
```

File: nats_rpc.go (hunk 2 above)
Comment (kozlovic) [inline nats_rpc.go]: Forgot about it. Will try.

**Comment by kozlovic** [comment]
Reverted to the use of a `nats.EncodedConn`. Updated PR with `push --force`

**Pull request merged**